proxy/http2/Http2ConnectionState.cc

```diff
@@ -229,6 +229,19 @@ rcv_headers_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
     }
   }
 
+  // [RFC 7540] 5.1 Stream States
+  // An endpoint MUST NOT send frames other than PRIORITY on a closed stream.
+  // An endpoint that receives any frame other than PRIORITY
+  // after receiving a RST_STREAM MUST treat that as a stream error.
+  if (stream->get_state() == Http2StreamState::HTTP2_STREAM_STATE_CLOSED) {
+    if (stream->recv_rst_stream == true) {
+      return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_STREAM, Http2ErrorCode::HTTP2_ERROR_STREAM_CLOSED,
+                        "recv headers stream closed");
+    } else {
+      return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_NONE);
+    }
+  }
+
```

Comment:
Seems like a straightforward solution.  However, I'm a bit unclear on how this scenario can happen.  Looking at rcv_reset_stream_frame(), it calls cstate.delete_stream() which starts the stream object on the process of shutting down but it does remove it from the connection state's stream_list.   That is the data structure that is used in cstate.find_stream().

So I'm unclear how the rcv_headers_frame() found a previously existing stream that had gone through rcv_headers_frame.

Reply:
Did you mean gone through rcv_reset_stream_frame() instead of "gone through rcv_headers_frame"?

I added some errors messages and it look like `stream->recv_rst_stream == false` all the time.